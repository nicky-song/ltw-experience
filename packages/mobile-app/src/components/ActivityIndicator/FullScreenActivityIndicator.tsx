import { StyleSheet, View } from 'react-native';
import { Text } from '@ant-design/react-native';
import { ActivityIndicator } from './ActivityIndicator';

export const FullScreenActivityIndicator = ({ text }: { text?: string }) => (
  <View style={styles.container}>
    {!!text && <Text style={styles.loadingText}>{text}</Text>}
    <ActivityIndicator />
  </View>
);

const styles = StyleSheet.create({
  container: {
    paddingTop: 150,
    height: '100%',
  },
  loadingText: {
    fontSize: 16,
    fontWeight: '400',
    marginBottom: 40,
    textAlign: 'center',
  },
});
