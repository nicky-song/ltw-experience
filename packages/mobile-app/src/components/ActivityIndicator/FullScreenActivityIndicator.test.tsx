import { render, screen } from '../../test/testing';
import { FullScreenActivityIndicator } from './FullScreenActivityIndicator';
import renderer from 'react-test-renderer';

describe('FullScreenActivityIndicator Component', () => {
  it('renders', () => {
    render(<FullScreenActivityIndicator />);
    expect(screen.getByTestId('activity-indicator')).toBeTruthy();
  });
  it('renders with text', () => {
    render(<FullScreenActivityIndicator text={'test'} />);
    expect(screen.getByText('test')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const activityIndicator1 = renderer.create(<FullScreenActivityIndicator />);
    expect(activityIndicator1).toMatchSnapshot();

    const activityIndicator2 = renderer.create(
      <FullScreenActivityIndicator text={'test'} />,
    );

    expect(activityIndicator2).toMatchSnapshot();
  });
});
