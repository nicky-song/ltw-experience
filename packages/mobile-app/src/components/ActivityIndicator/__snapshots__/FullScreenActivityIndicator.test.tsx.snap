// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenActivityIndicator Component matches snapshot 1`] = `
<View
  style={
    {
      "height": "100%",
      "paddingTop": 150,
    }
  }
>
  <ActivityIndicator
    color="#6FC07A"
    size="large"
    testID="activity-indicator"
  />
</View>
`;

exports[`FullScreenActivityIndicator Component matches snapshot 2`] = `
<View
  style={
    {
      "height": "100%",
      "paddingTop": 150,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "400",
        "marginBottom": 40,
        "textAlign": "center",
      }
    }
  >
    test
  </Text>
  <ActivityIndicator
    color="#6FC07A"
    size="large"
    testID="activity-indicator"
  />
</View>
`;
